### PR TITLE
Take in optional Overrides to Index-builder Config parameters

### PR DESCRIPTION
**Pull request by sbhaktha**
@row11 : here it is.

### CODE REVIEW

File: indexing/src/main/scala/org/allenai/common/indexing/BuildCorpusIndexRunner.scala [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
       c.copy(indexConfigName = x)
     } text ("name of configuration to use")
+    opt[File]("config-overrides-file") valueName ("<file>") action { (x, c) =>
+      c.copy(configOverrideFile = Some(x))
+    } text ("Config file with overrides if any")
   }
 
   parser.parse(args, IndexConfig()).foreach(config => {
     val rootConfig = ConfigFactory.parseResources(getClass, "indexing.conf").resolve()
+    val configOverrides = config.configOverrideFile map { f => ConfigFactory.parseFile(f) }
```
Comment (row11) [inline indexing/src/main/scala/org/allenai/common/indexing/BuildCorpusIndexRunner.scala]: I would personally put the pattern matching here that way you don't have to change the signature for `BuildCorpusIndex`. I mean something like:
```
val originalConfig = rootConfig.getConfig(s"org.allenai.common.indexing.${config.indexConfigName}")
val finalConfig = configOverrides match {
    case Some(overrides) => overrides.withFallback(originalConfig)
    case None => originalConfig
}
new BuildCorpusIndex(finalConfig.resolve()).buildElasticSearchIndex()
```
But up to you!

**Comment by row11** [comment]
One suggestion, otherwise LGTM!

File: indexing/src/main/scala/org/allenai/common/indexing/BuildCorpusIndex.scala [hunk 2]
```diff
@@ -32,7 +32,13 @@ import java.util.concurrent.TimeUnit
   * Refer http://joelabrahamsson.com/elasticsearch-101/ to get started.
   * Takes in Config object containing corpus and other information necessary to build the index.
   */
-class BuildCorpusIndex(config: Config) extends Logging {
+class BuildCorpusIndex(originalConfig: Config, configOverrides: Option[Config]) extends Logging {
```
Comment (row11) [inline indexing/src/main/scala/org/allenai/common/indexing/BuildCorpusIndex.scala]: if you do decide to leave this, maybe have `configOverrides` default to `None`?

**Comment by sbhaktha** [comment]
Good idea. Done!

**Comment by sbhaktha** [comment]
And Merging since it LGTY! :)

**Pull request merged**